ui/lib/sync/addon/routes/secrets.ts

```diff
@@ -9,23 +9,31 @@ import { hash } from 'rsvp';
 
 import type RouterService from '@ember/routing/router-service';
 import type StoreService from 'vault/services/store';
+import AdapterError from 'ember-data/adapter'; // eslint-disable-line ember/use-ember-data-rfc-395-imports
 
-interface ConfigResponse {
+interface ActivationFlagsResponse {
   data: {
-    disabled: boolean;
+    activated: Array<string>;
   };
 }
 
 export default class SyncSecretsRoute extends Route {
   @service declare readonly router: RouterService;
   @service declare readonly store: StoreService;
 
-  model() {
+  async model() {
+    const response = await this.store
+    .adapterFor('application')
+    .ajax('/v1/sys/activation-flags', 'GET')
+    .then(({ data: { activated } }: ActivationFlagsResponse) => {
+      activated.includes('secrets-sync');
+    }).catch((error: AdapterError) => {
+      return error;
+    })
+  
     return hash({
-      featureEnabled: this.store
-        .adapterFor('application')
-        .ajax('/v1/sys/sync/config', 'GET')
-        .then(({ data: { disabled } }: ConfigResponse) => !disabled),
+      featureEnabled: response.isAdapterError ? false : true,
```

Comment:
This needs to be updated to return the response, otherwise `featureEnabled` will always be true even if it's not actually enabled
```suggestion
      featureEnabled: response.isAdapterError ?  false : response,

Reply:
Following up here, after commenting [below](https://github.com/hashicorp/vault/pull/25739/commits/7d3f2289931533d31711f2485d530cfe39950e8d#r1515305488) - I think this should be the same as it was before and just return the response and error so we don't introduce any extra logic assigning true/false here